troubleshoot/ports/troubleshoot_ports.go

```diff
@@ -0,0 +1,34 @@
+package ports
+
+import (
+	"fmt"
+	"strings"
+)
+
+func TroubleshootDefaultPorts(host string) {
+	// Source - https://developer.hashicorp.com/consul/docs/install/ports
+	ports := []string{"8600", "8500", "8501", "8502", "8503", "8301", "8302", "8300"}
+	TroubleshootRun(ports, host)
+}
+
+func TroubleShootCustomPorts(host string, ports string) {
+	portsArr := strings.Split(ports, ",")
+	TroubleshootRun(portsArr, host)
+}
+
+func TroubleshootRun(ports []string, host string) {
+
+	resultsChannel := make(chan string)
+
+	var counter = 0
+
+	tcpTroubleShoot := TroubleShootTcp{}
+
+	for _, port := range ports {
```

Comment:
This is probably an UX related question. Should we wait for all the ports to be verified before printing the results? The current implementation prints the results as and when it appears. With syncGroups you could also remove the dependency of passing the channel to the `test` function. This way the test function just pings the port on the host and returns the `string` and the caller could add it to the channel. In this case, you could also convert resultsCh to a buffered channel equal to the length of ports. Something like


```go
for _, port := range ports {
    wg.Add(1)
    go func(p string){
         defer wg.Done()
         
         tcpTroubleshoot := &TroubleshootTCP{}
         res := tcpTroubleshoot.test(&hostPort{host: host, port: port})
         resultsCh <- res
    }
}

go func() {
   wg.Wait()
   close(resultsChannel)
}()

for _, res := range resultsCh {
    fmt.Println(res)
}
```

Thoughts?